### ISSUE TITLE
DBeaver: ora-01460 unimplemented or unreasonable conversion requested

### ISSUE THREAD

**Issue by sgabisaad** [serge-rider/dbeaver#1418]
I can connect to the database but when I try to open Tables/Views it gives the error.
[dbeaver-debug (2).txt](https://github.com/serge-rider/dbeaver/files/830269/dbeaver-debug.2.txt)

**Comment by serge-rider** [comment]
What Oracle version do you have?

During tables read DBeaver executes following query:
```sql
	SELECT /*+RULE*/ t.OWNER,t.TABLE_NAME as TABLE_NAME,'TABLE' as OBJECT_TYPE,'VALID' as STATUS,t.TABLE_TYPE_OWNER,t.TABLE_TYPE,t.TABLESPACE_NAME,t.PARTITIONED,t.IOT_TYPE,t.IOT_NAME,t.TEMPORARY,t.SECONDARY,t.NESTED,t.NUM_ROWS 
	FROM SYS.ALL_ALL_TABLES t
	WHERE t.OWNER='<SCHEMANAME>' AND NESTED='NO'
UNION ALL
	SELECT /*+RULE*/ o.OWNER,o.OBJECT_NAME as TABLE_NAME,'VIEW' as OBJECT_TYPE,o.STATUS,NULL,NULL,NULL,NULL,NULL,NULL,o.TEMPORARY,o.SECONDARY,NULL,0 
	FROM SYS.ALL_OBJECTS o 
	WHERE o.OWNER='<SCHEMANAME>' AND o.OBJECT_TYPE='VIEW'
```
Could you check how it works in your Oracle? (replace `<SCHEMANAME>` with real schema name)

**Comment by sgabisaad** [comment]
The Oracle version I am connecting to is: Oracle9i Enterprise Edition Release 9.2.0.8.0 - 64bit Production

I ran the query as suggested and attached are the results: It is a csv file, you can change the extension and check th results.

[SELECT_RULE_t_OWNER_t_TABLE_NAME_as_TABLE_NAME_TABLE_as_OBJECT_T_201703091407.txt](https://github.com/serge-rider/dbeaver/files/830758/SELECT_RULE_t_OWNER_t_TABLE_NAME_as_TABLE_NAME_TABLE_as_OBJECT_T_201703091407.txt)

**Comment by serge-rider** [comment]
Hmm. So this query was executed without error in SQL editor. But fails when DBeaver executes it internally.
This makes me a bit frustrating.

**Comment by sgabisaad** [comment]
Yes I can run the query but as soon as I try browsing to check the tables,... as highlighted in RED DOTS in the picture the error is displayed.

You can notice that I connected to the database and I can see the schema...

![capture png](https://cloud.githubusercontent.com/assets/26272375/23785801/1bac541a-0572-11e7-82cd-9a82140f03d1.jpg)

**Comment by serge-rider** [comment]
What JDBC driver do you use?

As far as I see from google this is a known issue with Oracle drivers.
http://www.dba-oracle.com/t_ora_01460_unimplemented_or_unreasonable_conversion_requested.htm
https://community.oracle.com/thread/3724859

Try to get another driver version (e.g. 11 or 12) from here: http://www.oracle.com/technetwork/database/features/jdbc/index-091264.html

**Comment by sgabisaad** [comment]
ojdbc7.jar

**Comment by sgabisaad** [comment]
I downloaded ojdbc8.jar now and tried still giving the same issue

**Comment by serge-rider** [comment]
Number in driver jar file is number of Java version, not driver version.

You can get real driver version (e.g. "11.2.0.3.0") in Properties view or from Test Connection dialog (from connection properties dialog).
As you have pretty old Oracle database (9i support ended many years ago) it makes sense to try different JDBC driver versions. 
9i (driver should be included in server distribution), 10, 11 and 12 (can be downloaded from link linked).

**Comment by sgabisaad** [comment]
Thanks for the soution. I created 3 different Oracle drivers based on the jdbc driver.

Note: The issue is resolved by using jdbc version 10.2.0.5. Attached is the 3 drivers I created which I will choose from when connecting to the database based on which version is the Host database I am connecting to.

![capture](https://cloud.githubusercontent.com/assets/26272375/23788088/b7c4511c-057d-11e7-975a-aa2e169aba65.PNG)
